### PR TITLE
invalidate the entire cloudflare zone when more than 50 files have ch…

### PR DESCRIPTION
**Pull request by manuelmeurer**
…anged

As discussed in #8

### CODE REVIEW

**Comment by manuelmeurer** [comment]
I'm not happy with the option name `invalidate_zone_for_many_files`. If anyone has a better name, I'm all :ear:s.

**Comment by ashfurrow** [comment]
The name makes sense to me :100:

File: spec/lib/middleman-cdn/cdns/cloudflare_spec.rb [hunk 1]
```diff
@@ -73,20 +96,45 @@
...
+          end
+          subject.invalidate(options, files)
         end
+      end
 
-        it "should not invalidate individual files" do
-          expect(double_cloudflare).not_to receive(:zone_file_purge)
-          subject.invalidate(options, files, all: true)
+      context "more than 50 files to invalidate" do
+        let(:files) { super() + ["/index.html"] }
```
Comment (leighmcculloch) [inline spec/lib/middleman-cdn/cdns/cloudflare_spec.rb]: Woah. TIL. I had no idea you could use `super` within `let`s. This. Is. Awesome.

**Comment by leighmcculloch** [comment]
The name makes sense to me. This looks great, thanks! There's an odd build failure only on ruby2.1.5.

File: spec/lib/middleman-cdn/cdns/cloudflare_spec.rb (hunk 1 above)
Comment (manuelmeurer) [inline spec/lib/middleman-cdn/cdns/cloudflare_spec.rb]: :smile: 
I had to look it up myself: https://github.com/rspec/rspec-core/commit/c5172be7378fcadf2b468c6d0ed367bb7471985c#diff-112a3ec80f232a6de7289060604874f6R39

**Comment by manuelmeurer** [comment]
Could you rerun the build for Ruby 2.1.5? I tried it locally and it works. Maybe a temporary error on Travis...

**Comment by manuelmeurer** [comment]
I removed one unnecessary `allow` from the specs, so the build was triggered again. Let's see if 2.1.5 works this time.

**Comment by manuelmeurer** [comment]
[Seems like it's a Bundler issue on Travis](https://github.com/rubygems/rubygems/issues/1220)... I updated the Travis config to install the latest Bundler version before running the specs.

**Comment by leighmcculloch** [comment]
Nice! Will merge and put this in a version shortly.

**Comment by manuelmeurer** [comment]
Is there an ETA for the new version? :wink:

**Comment by leighmcculloch** [comment]
@manuelmeurer Sorry I didn't come back to it! Thanks for the reminder. Merged and pushed in version `0.1.15`.

File: lib/middleman-cdn/cdns/cloudflare.rb [hunk 2]
```diff
@@ -15,7 +17,8 @@ def self.example_configuration_elements
           client_api_key: ['"..."', "# default ENV['CLOUDFLARE_CLIENT_API_KEY']"],
           email: ['"..."', "# default ENV['CLOUDFLARE_EMAIL']"],
           zone: ['"..."', ""],
-          base_urls: [['http://example.com', 'https://example.com'], ""]
+          base_urls: [['http://example.com', 'https://example.com'], ""],
+          invalidate_zone_for_many_files: [true, "# default true"]
```
Comment (ashfurrow) [inline lib/middleman-cdn/cdns/cloudflare.rb]: This says `true` by default, but I found it wasn't working until I explicitly added it to my config. Should I hope an issue?

Comment (leighmcculloch) [inline lib/middleman-cdn/cdns/cloudflare.rb]: Doh. Yeah please do. Thanks for reporting. Any idea why?

Comment (ashfurrow) [inline lib/middleman-cdn/cdns/cloudflare.rb]: Haven't looked into it yet.

**Pull request merged**